### PR TITLE
sync up worker status all the time

### PR DESCRIPTION
**Pull request by hougangliu**
Now, when metricsCollector for a worker failed to be created, the worker job status in studyjob.status.trials will not be updated (blocked in Running), and the corresponding job will not be deleted even completed.

Fixes: #298

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/299)

### CODE REVIEW

File: pkg/controller/studyjob/studyjob_controller.go [hunk 1]
```diff
@@ -348,57 +348,47 @@ func (r *ReconcileStudyJobController) updateWorker(c katibapi.ManagerClient, ins
...
 					cjob.Spec.Suspend = &susp
 					if err := r.Update(context.TODO(), cjob); err != nil {
 						return false, err
 					}
-
-					cwids = append(cwids, wid)
 				}
 			}
+		} else {
+		// for some reason, metricsCollector for this worker cannot be found (deleted by anyone accidentally or even failed to be created)
```
[the remark this answers is not in the document]
Comment (hougangliu) [reply pkg/controller/studyjob/studyjob_controller.go]: in this case, metrics including objectiveMetrics will not be collected, studyJob will have no idea which group of hyperParameter have better performance. That is, studyJob will study nothing

File: pkg/controller/studyjob/studyjob_controller.go [hunk 2]
```diff
@@ -348,57 +348,47 @@ func (r *ReconcileStudyJobController) updateWorker(c katibapi.ManagerClient, ins
...
 					susp := true
 					cjob.Spec.Suspend = &susp
 					if err := r.Update(context.TODO(), cjob); err != nil {
 						return false, err
 					}
-
-					cwids = append(cwids, wid)
 				}
 			}
+		} else {
```
[the remark this answers is not in the document]
Comment (hougangliu) [reply pkg/controller/studyjob/studyjob_controller.go]: it works well (I had verified it). `else` match with line 347 (not 348)

File: pkg/controller/studyjob/studyjob_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hougangliu) [reply pkg/controller/studyjob/studyjob_controller.go]: When a worker job changes to "Running", metric collector cronjob is spawned rightly. That is, when a worker job changes to "Complete", but its corresponding metric collector cronjob cannot be found. In this case, I think it is deleted by anyone accidentally or even failed to be created.

**Comment by richardsliu** [comment]
Thanks for your explanations.
/lgtm
/approve

**Pull request merged**